File: back/app/schemas/reviews.py
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreateRequest(BaseModel):
    car_id: int
    car_owner_id: int
    client_id: int | None = None
    rating: int = Field(..., ge=1, le=5)
    comment: str | None = None


class ReviewResponse(BaseModel):
    id: int
    car_id: int
    car_owner_id: int
    client_id: int | None
    rating: int
    comment: str | None
    create_date: datetime

    class Config:
        from_attributes = True

File: front/types/reviews.ts
export interface ReviewCreateRequest {
  car_id: number;
  car_owner_id: number;
  user_id?: number;
  rating: number; // 1-5
  comment?: string;
}

export interface ReviewResponse {
  id: number;
  car_id: number;
  car_owner_id: number;
  user_id?: number;
  rating: number;
  comment?: string;
  create_date: string;
}
